components/director/internal/domain/scenarioassignment/engine.go

```diff
@@ -28,3 +38,45 @@ func (engine) RemoveAssignedScenarios(in []*model.AutomaticScenarioAssignment) e
 	// remove scenarios from runtimes, which have label matching selector
 	return nil
 }
+
+func (s engine) GetScenariosForSelectorLabels(ctx context.Context, labels map[string]interface{}) ([]string, error) {
```

Comment:
Hmm, what do you think about the following name `ListScenariosFromSelectorLabels` or `ComputeScenariosFromSelectorLabels`. I think it's more clear what it does.

However, for me the issue is also in the `SelectorLabels` name.  I needed to read the method to know what it does. But unfortunatelyI don't have better idea for that.